packages/skills/ml_train/handlers.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the handler for the 'gym' skill."""
+import logging
+import sys
+from typing import cast, TYPE_CHECKING, Optional, List, Dict
+
+import numpy as np
+
+from aea.configurations.base import ProtocolId
+from aea.decision_maker.messages.transaction import TransactionMessage
+from aea.helpers.dialogue.base import DialogueLabel
+from aea.protocols.base import Message
+from aea.protocols.oef.message import OEFMessage
+from aea.protocols.oef.models import Description
+from aea.skills.base import Handler
+
+if TYPE_CHECKING or "pytest" in sys.modules:
+    from packages.protocols.ml_trade.message import MLTradeMessage
+    from packages.protocols.ml_trade.serialization import MLTradeSerializer
+    from packages.skills.ml_train.strategy import Strategy
+    # from packages.skills.ml_train.tasks import MLTask
+else:
+    from ml_trade_protocol.message import MLTradeMessage
+    from ml_trade_protocol.serialization import MLTradeSerializer
+    from ml_train_skill.strategy import Strategy
+    # from gym_skill.tasks import GymTask
+
+logger = logging.getLogger("aea.ml_train_skill")
+
+
+class TrainHandler(Handler):
+    """Gym handler."""
+
+    SUPPORTED_PROTOCOL = "ml_trade"
+
+    def __init__(self, **kwargs):
+        """Initialize the handler."""
+        logger.info("TrainHandler.__init__: arguments: {}".format(kwargs))
```

Comment:
would do debug rather than info level > less clutter.

Info should only be useful stuff: like received CFT etc.